### PR TITLE
[JENKINS-48447] Fixed HTTP 404 error when clicking on newView sidebar link from an other view.

### PR DESCRIPTION
**Pull request by Jochen-A-Fuerbacher**
See [JENKINS-48447](https://issues.jenkins-ci.org/browse/JENKINS-48447).

### Proposed changelog entries

* Fixed HTTP 404 error when clicking on newView sidebar link from an other view.

### Submitter checklist

- [x] JIRA issue is well described
- [x] Changelog entry appropriate for the audience affected by the change (users or developer, depending on the change).
- [x] Appropriate autotests or explanation to why this change has no tests

### CODE REVIEW

**Review by oleg-nenashev — CHANGES_REQUESTED** [review]
`static final String URL_NAME = Jenkins.getInstance().getRootUrl()` is dangerous, because we do not know when the object will be initialized. If Jenkins has not been started by that time, `getInstance()` will fail. The static variable also will not reflect changes in the configuration by the user, restart will be needed

**Comment by daniel-beck** [comment]
I think this will also not handle changes of the root URL well.

**Comment by Jochen-A-Fuerbacher** [comment]
Thanks for your feedback. This makes sense. I changed it to get the root url each time `getUrlName()` gets called.

File: core/src/main/java/jenkins/model/NewViewLink.java [hunk 1]
```diff
@@ -40,7 +38,8 @@ public String getDisplayName() {
 
             @Override
             public String getUrlName() {
-                return URL_NAME;
+            	String urlName = Jenkins.getInstance().getRootUrl() + "newView";
```
Comment (oleg-nenashev) [inline core/src/main/java/jenkins/model/NewViewLink.java]: NIT: indenting

File: core/src/test/java/jenkins/model/NewViewLinkTest.java [hunk 2]
```diff
@@ -27,7 +27,6 @@ public void getActionsHasPermission() throws Exception {
         final Action action = actions.get(0);
         assertEquals(Messages.NewViewLink_NewView(), action.getDisplayName());
         assertEquals(NewViewLink.ICON_FILE_NAME, action.getIconFileName());
-        assertEquals(NewViewLink.URL_NAME, action.getUrlName());
```
Comment (oleg-nenashev) [inline core/src/test/java/jenkins/model/NewViewLinkTest.java]: I would rather rewrite the assert to check the full URL or to chack whether the URL contains the name. Just removing the check is a bit hackish

File: core/src/main/java/jenkins/model/NewViewLink.java [hunk 3]
```diff
@@ -14,7 +14,7 @@
     @VisibleForTesting
     static final String ICON_FILE_NAME = "folder";
     @VisibleForTesting
-    public static final String URL_NAME = "newView";
+    static final String URL_NAME_SUFFIX = "newView";
```
Comment (oleg-nenashev) [inline core/src/main/java/jenkins/model/NewViewLink.java]: formally it is a compat-breaking change. I would recommend not to do so

Comment (daniel-beck) [reply core/src/main/java/jenkins/model/NewViewLink.java]: Should probably have been `@Restricted` to begin with, AFAIUI `@VisibleForTesting` serves the same purpose, just not enforcing it.

Comment (daniel-beck) [reply core/src/main/java/jenkins/model/NewViewLink.java]: There's no reason the entire class shouldn't be `@Restricted`.

**Review by oleg-nenashev — APPROVED** [review]
LGTM

**Comment by jglick** [comment]
Amends #3086.

**Comment by daniel-beck** [comment]
Will fail with nested views?

**Comment by jglick** [comment]
Perhaps, but more urgently, it fails with folders.

**Comment by daniel-beck** [comment]
Oops, merged 2y 1w ago, misread as 1wk…

**Pull request merged**